app/services/weights_and_measures.rb

```diff
@@ -45,8 +45,10 @@ def system
     }
   }.freeze
 
-  def scales_for_variant_unit
-    @scales_for_variant_unit ||= @units[@variant.product.variant_unit]&.reject { |_scale, unit_info|
+  def scales_for_variant_unit(ignore_available_units: false)
+    return @units[@variant.product.variant_unit] if ignore_available_units
+
+    @units[@variant.product.variant_unit]&.reject { |_scale, unit_info|
```

Comment:
This no longer caches the result, but I assume caching wasn't needed anyway. Even if it gets called twice, it seems like an inexpensive process.